spec/logr/entry_spec.rb

```diff
@@ -0,0 +1,118 @@
+require 'spec_helper'
+
+describe Logr::Entry do
+  let(:logger) { double('logger') }
+  let(:entry) { Logr::Entry.new(logger) }
+
+  describe '#event' do
+    let(:entry_event) { entry.event('test_event', context_1_key: 'context_1_value') }
+
+    it 'should add the event with the correct name' do
+      expect(entry_event.instance_variable_get(:@event).name).to eq('test_event')
+    end
+
+    it 'should add the event with the correct context' do
+      expect(entry_event.instance_variable_get(:@event).context).to include(context_1_key: 'context_1_value')
```

Comment:
I think it is not a problem here to reach into `Event`. We that `Entry` has set up the event correctly based on the inputs.

You cannot use the `attr_reader` macro here, but you can roll your own reader method:

```ruby
class Entry
  def get_event
    @event
  end
end
```

I'm not sure on the name tho.